Modules/Scripted/VPAWModel/VPAWModel.py

```diff
@@ -377,58 +414,287 @@ def __init__(self):
+        )
 
-        import time
+    def convertFCSVLandmarksToP3(self, vPAWRootDirectory):
+        import conversion_utils.generate_pixel_space_landmarks
 
-        startTime = time.time()
-        logging.info("Processing started")
-
-        # Compute the thresholded output volume using the "Threshold Scalar Volume" CLI
-        # module
-        cliParams = {
-            "InputVolume": inputVolume.GetID(),
-            "OutputVolume": outputVolume.GetID(),
-            "ThresholdValue": imageThreshold,
-            "ThresholdType": "Above" if invert else "Below",
-        }
-        cliNode = slicer.cli.run(
-            slicer.modules.thresholdscalarvolume,
-            None,
-            cliParams,
-            wait_for_completion=True,
-            update_display=showResult,
+        conversion_utils.generate_pixel_space_landmarks.convert_landmarks(
+            os.path.join(vPAWRootDirectory, "images"),
+            os.path.join(vPAWRootDirectory, "landmarks"),
+            os.path.join(vPAWRootDirectory, "transformed_landmarks"),
         )
-        # We don't need the CLI module node anymore, remove it to not clutter the scene
-        # with it
-        slicer.mrmlScene.RemoveNode(cliNode)
 
-        stopTime = time.time()
-        logging.info(f"Processing completed in {stopTime-startTime:.2f} seconds")
+    def runSegmentation(self, vPAWRootDirectory, vPAWModelsDirectory):
+        cwd = os.getcwd()
+        ConfigHandle, ConfigName = None, None
+        SegmentHandle, SegmentName = None, None
+        try:
+            # Change directory and create files
+            os.chdir(self.pediatric_airway_atlas_directory)
+            ConfigHandle, ConfigName = tempfile.mkstemp(suffix=".yaml", text=True)
+            SegmentHandle, SegmentName = tempfile.mkstemp(suffix=".yaml", text=True)
+
+            # Add text to the main configuration file
+            b_s_f_s = "False"
+            n_p = 1
+            os.write(
+                ConfigHandle,
+                f"""root: {vPAWRootDirectory}
+n_samples: -1  # how many samples to use to build the atlas, -1 denotes to use all
+metadata_excel_fname: FilteredControlBlindingLogUniqueScanFiltered.xls
+band_depth_ages: [20, 40, 60, 80, 100, 120, 140]  # also for functional boxplot
+n_centerline_points: 200
+n_curve_points: 500
+TARGET_LANDMARKS_ORDERED: [nasalspine, choana, epiglottistip, tvc, subglottis, carina]
+dilation_times: 20
+ALL_POSSIBLE_LANDMARKS: [carina, tracheacarina, trachea, tvc, subglottis, epiglottistip, columella, nasalspine, rightalarim, leftalarim, nosetip, [choana,posteriorinferiorvomercorner], pyrinaaperture, [baseoftongue,tonguebase]]
+plane_estimation_based_mesh_area: True
+use_planes_for_laplace_marking: True
+balance_spacing_for_segmentation: {b_s_f_s}
+num_processes: {n_p}
+""".encode(),
+            )
+            os.close(ConfigHandle)
+            ConfigHandle = None
+
+            # Add text to the configuration file for segmentation
+            os.write(
```

Comment:
Same here, I would say read and write yaml using pyyaml

Reply:
Will do.